commcare/src/main/java/org/motechproject/commcare/events/imports/ImportFormActionEventHandler.java

```diff
@@ -0,0 +1,65 @@
+package org.motechproject.commcare.events.imports;
+
+import org.joda.time.DateTime;
+import org.motechproject.commcare.events.constants.EventSubjects;
+import org.motechproject.commcare.pull.CommcareFormImporterImpl;
+import org.motechproject.commcare.service.CommcareFormService;
+import org.motechproject.commons.api.Range;
+import org.motechproject.event.MotechEvent;
+import org.motechproject.event.listener.EventRelay;
+import org.motechproject.event.listener.annotations.MotechListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+
+import static org.motechproject.commcare.events.constants.EventDataKeys.END_DATE;
+import static org.motechproject.commcare.events.constants.EventDataKeys.START_DATE;
+
+/**
+ * Class responsible for handling "Import Forms" action in tasks.
+ */
+@Component
+public class ImportFormActionEventHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ImportFormActionEventHandler.class);
+
+    @Autowired
+    private EventRelay eventRelay;
+
+    @Autowired
+    private CommcareFormService formService;
+
+    private CommcareFormImporterImpl importer;
+
+    @PostConstruct
+    public void init() {
+        importer = new CommcareFormImporterImpl(eventRelay, formService);
```

Comment:
We will need to create it per event. Let's create a separate factory that will return instances of the importer.